From e1d0edc04ff9a24faa29cd9aa964d889a2b8176b Mon Sep 17 00:00:00 2001
From: Max Krummenacher <max.oss.09@gmail.com>
Date: Thu, 14 May 2026 12:26:30 +0200
Subject: [PATCH] milkv_duo: mainline: use fat for env and use the default env
 txt file

and reduce MALLOC LEN to 0x80000
and remove spi flash
and add bootcmd

Upstream-Status: Pending
Signed-off-by: Max Krummenacher <max.oss.09@gmail.com>
---
 configs/milkv_duo_defconfig | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/configs/milkv_duo_defconfig b/configs/milkv_duo_defconfig
index f1f3930564bf..d9f7571dedb2 100644
--- a/configs/milkv_duo_defconfig
+++ b/configs/milkv_duo_defconfig
@@ -1,9 +1,10 @@
 CONFIG_RISCV=y
-CONFIG_SYS_MALLOC_LEN=0x820000
+CONFIG_SYS_MALLOC_LEN=0x80000
 CONFIG_SYS_MALLOC_F_LEN=0x8000
 CONFIG_NR_DRAM_BANKS=1
 CONFIG_HAS_CUSTOM_SYS_INIT_SP_ADDR=y
 CONFIG_CUSTOM_SYS_INIT_SP_ADDR=0x82300000
+CONFIG_ENV_SIZE=0x2000
 CONFIG_DEFAULT_DEVICE_TREE="cv1800b-milkv-duo"
 CONFIG_SYS_LOAD_ADDR=0x80080000
 CONFIG_IDENT_STRING="milkv_duo"
@@ -12,6 +13,7 @@ CONFIG_ARCH_RV64I=y
 CONFIG_RISCV_SMODE=y
 CONFIG_FIT=y
 CONFIG_SUPPORT_RAW_INITRD=y
+CONFIG_USE_BOOTCOMMAND=y
 CONFIG_SYS_CBSIZE=512
 CONFIG_SYS_PBSIZE=544
 CONFIG_HUSH_PARSER=y
@@ -24,6 +26,9 @@ CONFIG_CMD_PXE=y
 CONFIG_CMD_FAT=y
 CONFIG_CMD_FS_GENERIC=y
 CONFIG_ENV_OVERWRITE=y
+CONFIG_ENV_USE_DEFAULT_ENV_TEXT_FILE=y
+CONFIG_ENV_DEFAULT_ENV_TEXT_FILE="board/sophgo/milkv_duo/milkv-duo.env"
+CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
 CONFIG_NET_RANDOM_ETHADDR=y
 CONFIG_CLK_SOPHGO_CV1800B=y
 CONFIG_MMC=y
@@ -32,11 +37,8 @@ CONFIG_MMC_UHS_SUPPORT=y
 CONFIG_MMC_SDHCI=y
 CONFIG_MMC_SDHCI_ADMA=y
 CONFIG_MMC_SDHCI_CV1800B=y
-CONFIG_SPI_FLASH_MACRONIX=y
 CONFIG_ETH_DESIGNWARE=y
 CONFIG_SYS_NS16550=y
 CONFIG_SYS_NS16550_MEM32=y
-CONFIG_SPI=y
-CONFIG_CV1800B_SPIF=y
 CONFIG_SYSRESET=y
 CONFIG_SYSRESET_CV1800B=y
-- 
2.51.0

